### PR TITLE
GAWB-1898: Prevent XSS in catalog wizard

### PR DESCRIPTION
**Pull request by MattPutnam**
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Squash commits and merge to develop
- [x] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by davidangb** [comment]
Some thoughts:
1. Have you compared jQuery `.text()` vs. `clojure.string/escape`? If `escape` is just as good, it would be a simpler implementation.
2. are we also vulnerable at library_page.cljs line 183 (at `:typeaheadSuggestionTemplate`)?
3. I really want to see some tests for this - unit tests that compare raw vs. escaped strings and/or browser tests that check for XSS injections.

**Comment by MattPutnam** [comment]
library_page.cljs seems fine.

File: src/cljs/main/broadfcui/page/library/library_page.cljs [hunk 1]
```diff
@@ -181,7 +181,8 @@
         :typeaheadDisplay (fn [result]
                             (.text (js/$ (str "<div>" (aget result "value") "</div>"))))
         :typeaheadSuggestionTemplate (fn [result]
-                                       (str "<div style='textOverflow: ellipsis; overflow: hidden; font-size: smaller;'>" (aget result "value") "</div>"))}]])})
+                                       (-> (js/$ "<div style='textOverflow: ellipsis; overflow: hidden; font-size: smaller;'>")
+                                           (.append (aget result "value"))))}]])})
```
Comment (davidangb) [inline src/cljs/main/broadfcui/page/library/library_page.cljs]: use `text()`, not `append()`!

**Comment by MattPutnam** [comment]
Reducing scope to just the catalog wizard.  Will file a separate ticket for the library page, as this may require an orchestration change.

**Review by davidangb — APPROVED** [review]
👍 wfm!

**Review by ahaessly — APPROVED** [review]
i don't really understand what this is doing... but if it fixes the problem i'm ok with it.

**Pull request merged**